library/ssl_tls.c

```diff
@@ -3102,6 +3102,27 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
     }
 #endif
 
+#if !defined(DEPRECATED_REMOVED)
+    /* Heap allocate and translate curve_list from internal to IANA group ids */
+    if ( ssl->conf->curve_list ) {
+        size_t length;
+	for (length = 0; ssl->conf->curve_list[length] != MBEDTLS_ECP_DP_NONE; length++) {} 
+
+	uint16_t *group_list = mbedtls_calloc(length, sizeof(mbedtls_ecp_group_id));
+
+	for (size_t i = 0; ssl->conf->curve_list[i] != MBEDTLS_ECP_DP_NONE; i++) {
+            const mbedtls_ecp_curve_info *info = mbedtls_ecp_curve_info_from_grp_id(ssl->conf->curve_list[i]);
```

Comment:
We should handle unknown curves - either fail gracefully or ignore.

Reply:
Overly long line

Reply:
Comment on unknown curves is not yet addressed.

Reply:
Still unfixed

Reply:
Apologies for missing this. The for loop breaks when an unknown curve is encountered, treating it like a null-terminator.
The long line has been shortened.